haystack/nodes/prompt/invocation_layer/anthropic_claude.py

```diff
@@ -0,0 +1,229 @@
+    """
+
+    def __init__(self, api_key: str, model_name_or_path: str = "claude-v1", max_length=200, **kwargs):
+        """
+         Creates an instance of PromptModelInvocation Layer for Claude models by Anthropic.
+        :param model_name_or_path: The name or path of the underlying model.
+        :param max_tokens_to_sample: The maximum length of the output text.
+        :param api_key: The Anthropic API key.
+        :param kwargs: Additional keyword arguments passed to the underlying model. The list of Anthropic-relevant
+        kwargs includes: stop_sequences, temperature, top_p, top_k, and stream. For more details about these kwargs,
+        see Anthropic's [documentation](https://console.anthropic.com/docs/api/reference).
+        """
+        super().__init__(model_name_or_path)
+        if not isinstance(api_key, str) or len(api_key) == 0:
+            raise AnthropicError(
+                f"api_key {api_key} must be a valid Anthropic key. Visit https://console.anthropic.com/account/keys to get one."
+            )
+        self.api_key = api_key
+        self.max_length = max_length
+
+        # Due to reflective construction of all invocation layers we might receive some
+        # unknown kwargs, so we need to take only the relevant.
+        # For more details refer to Anthropic documentation
+        supported_kwargs = ["temperature", "top_p", "top_k", "stop_sequences", "stream", "stream_handler"]
+        self.model_input_kwargs = {k: v for (k, v) in kwargs.items() if k in supported_kwargs}
+
+        # Number of max tokens is based on the official Anthropic model pricing from:
+        # https://cdn2.assets-servd.host/anthropic-website/production/images/FINAL-PRICING.pdf
+        self.max_tokens_limit = 9000
+        self.tokenizer: Tokenizer = self._init_tokenizer()
+
+    def _init_tokenizer(self) -> Tokenizer:
+        # Expire cache after a day
+        expire_after = 60 * 60 * 60 * 24
+        # Cache the JSON config to avoid downloading it each time as it's a big file
+        with requests_cache.enabled(expire_after=expire_after):
+            res = request_with_retry(method="GET", url=CLAUDE_TOKENIZER_REMOTE_FILE)
+            res.raise_for_status()
+        return Tokenizer.from_str(res.text)
+
+    def invoke(self, *args, **kwargs):
+        """
+        Invokes a prompt on the model. It takes in a prompt and returns a list of responses using a REST invocation.
+        :return: The responses are being returned.
+        """
+
+        human_prompt = "\n\nHuman: "
+        assitant_prompt = "\n\nAssistant: "
+
+        prompt = kwargs.get("prompt")
+        if not prompt:
+            raise ValueError(
+                f"No prompt provided. Model {self.model_name_or_path} requires prompt."
+                f"Make sure to provide prompt in kwargs."
+            )
+
+        kwargs_with_defaults = self.model_input_kwargs
+
+        if "stop_sequence" in kwargs:
```

Comment:
We have stop_words as the parameter at the prompt node init level that we then pass to the invocation layer. And this invocation layer then translates "stop_words" to the specific parameter name in the invocation layer. As we stand here, but I am not 100% sure, it seems like stop _words from PromptNode level will not get translated to "stop_sequence" parameter. Please double check

Reply:
Update: it will get translated but double-check why the default case when no stop_words are passed from prompt node level text generation fails

Reply:
It should, but in this case, it is a problem because stop_words from PromptNode get passed as None because we allow None. And Anthropic, unlike all other providers, doesn't like None and barfs the exception.

Reply:
Ah, makes sense. I'll see to handle that case.